include/seqan3/search/detail/search_traits.hpp

```diff
@@ -92,6 +93,9 @@ struct search_traits
                                                      output_reference_id |
                                                      output_reference_begin_position |
                                                      output_index_cursor;
+
+    //!\brief A flag indicating whether a user provided callback was given.
+    static constexpr bool is_one_way_execution = search_configuration_t::template exists<search_cfg::on_result>();
```

Comment:
I am confused with the naming. "Two-way execution" I first got to know when you introduced it in the context of executors/parallelization (or do I remember that wrongly?). Now you use one-way execution for a user defined callback? Can we maybe rather use something like `has_user_defined_callback`? 

EDIT: I see below that you use it to differentiate between executors. Can you change the brief then? `If the user provides a callback, we need to use a one_way_execution...`

Reply:
This is the point. In the executor context two-way means that it returns something to the user he can wait for and one-way means it does not return anything but just executes it and the user needs to make sure the code is handled properly. But I see that the term might be confusing and I will rename it.